force-app/main/default/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -127,12 +128,14 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
         // Insert/Update any pending RecurringDonationSchedule records before the
         // evaluation service is called so that the full schedule is available to the
         // service to determine what Opportunities to insert or modify
-        TDTM_TriggerHandler.processDML(dmlActions);
+        TDTM_TriggerHandler.processDMLWithErrors(dmlActions);
         dmlActions = null;
 
         findMatchingElevateOpps(rds, triggerAction);
 
-        evaluateOpportunities(rds, oldRds, triggerAction);
+        if (TDTM_TriggerHandler.noErrorInCurrentTransaction(objResult, rds)) {
```

Comment:
Is this just here to prevent calling `evaluateOpportunities` twice? We still want to run this for the successful records.